bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/api/ViessmannApi.java

```diff
@@ -319,4 +303,29 @@ private Properties setHeaders() throws ViessmannAuthException {
         headers.put("Authorization", "Bearer " + atr.accessToken);
         return headers;
     }
+
+    private void handleViError(String response) {
+        ViErrorDTO viError = GSON.fromJson(response, ViErrorDTO.class);
+        if (viError != null) {
+            switch (viError.getStatusCode()) {
+                case HttpStatus.TOO_MANY_REQUESTS_429:
+                    logger.warn("ViError: {} | Resetting Limit at {}", viError.getMessage(),
+                            viError.getExtendedPayload().getLimitResetDateTime());
+                    bridgeHandler.updateBridgeStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            String.format("API Call limit reached. Reset at %s",
+                                    viError.getExtendedPayload().getLimitResetDateTime()));
+                    bridgeHandler.waitForApiCallLimitReset(viError.getExtendedPayload().getLimitReset());
+                    break;
+                case HttpStatus.BAD_GATEWAY_502:
+                    logger.debug("ViError: {} | Device not reachable", viError.getMessage());
+                    bridgeHandler.updateBridgeStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
```

Comment:
Do we really want to set the bridge to offline if a device is unreachable? Shouldn't the device go offline and not the bridge?

Reply:
Good point. I'll check that.

Reply:
For a proper status handling of thing handler I submitted https://github.com/smarthomej/addons/pull/428 to improve the polling.

Reply:
Done. See https://github.com/smarthomej/addons/pull/426/commits/69e576fd77447243e2f8b6dd5c7216bd1339b657.